phpBB/phpbb/db/migrator.php

```diff
@@ -488,18 +529,27 @@ protected function process_data_step($steps, $state, $revert = false)
 
 				// Set state to false since we reached the point we were at
 				$state = false;
+
+				// There is a tendency to get stuck in some cases
```

Comment:
If `dbtools.perform_schema_changes()` is called, this method will return an array, no matter what. That means, on the next call it will continue on the *same* step. Thus, we have to move forward if the last call succeeded and "not get stuck".
I will also fix this comment, then.